backend/prisma/schema.prisma

```diff
@@ -23,9 +23,37 @@ model User {
   email String   @unique
   type  UserType
 
-  classes Class[]
-  lessons Lesson[]
-  tasks   Task[]
+  keyPair              KeyPair?
+  authenticationTokens AuthenticationToken[]
+  classes              Class[]
+  lessons              Lesson[]
+  tasks                Task[]
+}
+
+model KeyPair {
+  id                   Int      @id @default(autoincrement())
+  teacherId            Int      @unique
```

Comment:
fragmentation ;)

Reply:
Actually there was another reason I just remembered when I started addressing the issue: This allows for some performance optimizations (in the future) where we don't need to try decrypting all values with all keys - we already know it will fail if the ids don't match.